.github/workflows/acquia-backup-databases.yml

```diff
@@ -0,0 +1,99 @@
+        description: 'The client secret for authenticating with ACLI.'
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  backup-databases:
+    name: 'Backup Acquia databases'
+    runs-on: ubuntu-latest
+    outputs:
+      databases: ${{ steps.get_databases.outputs.database_list }}
+      db_list: ${{ steps.get_databases.outputs.db_list }}
+    steps:
+      - name: Checking out code
+        uses: actions/checkout@v3
+
+      - name: Installing PHP
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: ${{ inputs.php_version }}
+
+      - name: Installing dependencies
+        run: |
+          curl -OL https://github.com/acquia/cli/releases/download/2.13.0/acli.phar
+          echo "ACLI downloaded!"
+
+      - name: Authorizing with ACLI
+        run: |
+          php acli.phar auth:login -n \
+            --key="${{ secrets.ACLI_CLIENT_ID }}" \
+            --secret="${{ secrets.ACLI_CLIENT_SECRET }}"
+
+      - name: Retrieving all database names for this environment
+        id: get_databases
+        # Gets databases filtering by a name. If no filter is provided, it'll
+        # retrieve all databases.
+        run: |
+          # The first time we run api:environments:database-list acli very
+          # kindly also outputs a string asking the user to share
+          # anonymous user data along with the JSON output causing an error
+          # further down the chain - this simply mitigates that.
+          acsf_request=$(php acli.phar api:environments:database-list \
+            --filter='${{ inputs.filter }}' \
+            --limit='1' \
+            ${{ secrets.ENVIRONMENT_ID }})
+
+          # Gets a list of databases and uses jq to filter the output
+          # to the name value only.
+          database_list=$(php acli.phar api:environments:database-list \
+            --filter='${{ inputs.filter }}' \
+            ${{ secrets.ENVIRONMENT_ID }} \
+            | jq '.[] | .name')
+
+          # Processes the output of databases to allow the data to
+          # be passed to the 'Backing up databases' step.
+          delimiter="$(openssl rand -hex 8)"
+          echo "db_list<<${delimiter}" >> "${GITHUB_OUTPUT}"
+          echo ${database_list} >> ${GITHUB_OUTPUT}
+          echo "${delimiter}" >> "${GITHUB_OUTPUT}"
+
+      - name: Backing up databases
+        run: |
+
+          # Split db_list into an array.
+          IFS=', ' read -r -a backup_array <<< "${{ steps.get_databases.outputs.db_list }}"
+
+          # Loop through the array of databases and attempt to 
+          # perform a backup for each database.
+          i=0
+          while [ $i -lt ${#backup_array[@]} ]
+          do
+            php acli.phar api:environments:database-backup-create ${{ secrets.ENVIRONMENT_ID }} ${backup_array[$i]}
+            echo "💧 Successfully backed up database "${backup_array[$i]}"" || echo "❌ Error backing up database "${backup_array[$i]}". See logs."
```

Comment:
```suggestion
            php acli.phar api:environments:database-backup-create \
              --no-interaction \
              ${{ secrets.ENVIRONMENT_ID }} \
              ${backup_array[$i]} \
              && echo "💧 Successfully backed up database "${backup_array[$i]}"" \
              || echo "❌ Error backing up database "${backup_array[$i]}". See logs."
```

I mean you could be tidier with this still :joy: but just to be in keeping with other commands (where it's split across multiple lines), _and also_ remember that the syntax is `command && success_message || fail_message`

With the echos on the line at the moment you would probably never see the error output - it'd only print that if there was an error echoing the string you've passed in to the first echo command. (it's saying, `echo X || if that fails echo Y`.)

https://unix.stackexchange.com/a/88851

Reply:
More accurately (reading from that link), 

`|| X` will run if the argument to the left is false
`&& Y` will run if the argument to the left is true

`command && X` will run X if command succeeds/is true, and nothing else will run after that
`command && X || Y` will run Y if command fails/is false, in which case X won't be evaluated, and Y will run instead 

alternatively,
`command || X` will run X if command fails/is false
`command || X && Y` will run Y if command succeeds/is true, OR if command fails/is false BUT X succeeds/is true.

😬 make sense? I'm not sure if that makes sense.